app/models/mixins/assignment_mixin.rb

```diff
@@ -3,6 +3,17 @@
 module AssignmentMixin
   extend ActiveSupport::Concern
   ESCAPED_PREFIX = "escaped".freeze
+  NAMESPACE_SUFFIX = "assigned_to".freeze
+
+  # This module method was not intended to be mixed into a class. It is used outside
+  # of any class that may mix this in, (Classification) to find the assignments associated
+  # with a given tag.
```

Comment:
Wait wat? I don't understand; Why is this not just then a scope on `Tag`, as that's exactly what it does? 

Actually, just to match the current convention that I'm seeing, why isn't it at last just included in the `ClassMethods` belong to be included on the class? Then it'd match the current `.assignments` method.

Reply:
Chris, per your second point, I wanted to do it that way. But, the code in `Classification` that needs to delete these tags has no way of knowing which classes have included the mixin and would not know which classes on which to call the method. 

I didn't consider making it a scope on `Tag`. I think that would work fine. But, then knowledge of how assignments are using tags would be added to the `Tag` class and I think it should stay in the mixin.

Reply:
Ahh I see, I thought that `Classification` included this mixin.

In that case, you can make your intentions clear and without your comment by doing the following:

```ruby
def all_assignments(tag = nil)
  # ...
end
module_function :all_assignments
```

This will signal that this method is intended to be called with the mixin itself as the receiver. It will be mixed in to instances of the class, but will be automatically marked as private. (https://ruby-doc.org/core-2.4.2/Module.html#method-i-module_function)